File: module6/sprint2/book_project/be/book/src/main/java/com/example/entity/Book.java
package com.example.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String bookName;
    private String description;
    private Double price;
    private String image;
    private Integer pageNumber;
    private String authors;
    private String publishingHouse;
    @Column(columnDefinition = "BIT(1) DEFAULT 0")
    private Boolean isDeleted;

    @ManyToOne()
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    private Category category;
}

File: module6/sprint2/book_project/fe/book - Copy (2) - Copy/src/app/model/book.ts
import {Category} from './category';

export interface Book {
  id?: number;
  bookName?: string;
  description?: string;
  price?: number;
  image?: string;
  pageNumber?: number;
  authors?: string;
  publishingHouse?: string;
  quantity?: number;
  category?: Category;
  isDeleted?: boolean;
}
